dist/modularity/init.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -ueo pipefail
+
+VENV_PATH=".venv"
+[[ -f .venv/bin/activate ]] || virtualenv -q -p python3 .venv
+chmod 755 .venv/bin/activate
+. "$VENV_PATH"/bin/activate
+
+REPO_URL="https://github.com/wiredtiger/layercparse.git"
+BRANCH="main"
+IS_CACHED=$(pip3 show layercparse > /dev/null 2>&1 && echo true || echo false)
+CACHE_TIMEOUT=0.5
+LATEST_HASH_CMD="git ls-remote \"$REPO_URL\" \"$BRANCH\" | awk '{print \$1}'"
+if ($IS_CACHED); then
+    if ! LATEST_HASH=$(timeout "$CACHE_TIMEOUT"s bash -c "$LATEST_HASH_CMD"); then
```

Comment:
`timeout` doesn't exist on MacOS. I suggest implementing this in bash.

Also, if we hit timeout, we still want to run a background update.

And can we use `stat` or something to compare timestamps so that we don't have to go to internet each time when we run it? There could be a minimal interval of 1 hour or 1 day or so.

```bash
#!/bin/bash

REPO_URL="https://github.com/wiredtiger/layercparse.git"
BRANCH="main"
TIMEOUT=0.5

is_cached() {
  pip3 -qq show layercparse
  # true # for debug
}

get_latest_hash() {
  #sleep 2 # for debug
  git ls-remote "$REPO_URL" "$BRANCH" | awk '{print $1}'
}

get_latest_hash_timed() {
  get_latest_hash &
  P1="$!"
  (
    sleep $TIMEOUT
    kill $P1 2>/dev/null
  ) &
  wait $P1
}

if is_cached; then
  echo "Cached"
  if HASH=`get_latest_hash_timed`; then
    echo "Got hash: $HASH"
  else
    if [[ $? -eq 143 ]]; then
      echo "Timeout. Getting hash in the background"
      (
        HASH=`get_latest_hash`
        echo "Background hash = $HASH"
        # ... update the environment
      ) 2>&1 > /dev/null &
    else
      echo "Failed to get hash"
    fi
  fi
else
  echo "Not cached"
  HASH=`get_latest_hash`
  echo "Hash: $HASH"
fi
```

Reply:
Updated in [371f6e7](https://github.com/wiredtiger/wiredtiger/pull/11335/commits/371f6e705827a8e4d9b4e169dc8a096a4e3fad65)